[javac] NPE while computing quickfixes

In a class implementing Serializable, but without serialVersionUID, like:
```java
class MyClass implements Serializable {}
```

 if you select the "Add generated serial version ID" quickfix, an NPE occurs:
```
[Error - 4:09:54 PM] Request codeAction/resolve failed.
  Message: Internal error.
  Code: -32603 
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "org.eclipse.jdt.core.dom.ASTNode.getStartPosition()" because "node" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:649)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1491)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:2073)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2035)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)
Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.jdt.core.dom.ASTNode.getStartPosition()" because "node" is null
	at org.eclipse.jdt.core.dom.rewrite.ASTRewrite.getRootNode(ASTRewrite.java:339)
	at org.eclipse.jdt.core.dom.rewrite.ASTRewrite.rewriteAST(ASTRewrite.java:275)
	at org.eclipse.jdt.internal.corext.refactoring.structure.CompilationUnitRewrite.attachChange(CompilationUnitRewrite.java:290)
	at org.eclipse.jdt.internal.corext.refactoring.structure.CompilationUnitRewrite.createChange(CompilationUnitRewrite.java:252)
	at org.eclipse.jdt.internal.corext.fix.CompilationUnitRewriteOperationsFixCore.createChange(CompilationUnitRewriteOperationsFixCore.java:126)
	at org.eclipse.jdt.internal.ui.text.correction.proposals.FixCorrectionProposalCore.createTextChange(FixCorrectionProposalCore.java:97)
	at org.eclipse.jdt.core.manipulation.CUCorrectionProposalCore.createChange(CUCorrectionProposalCore.java:195)
	at org.eclipse.jdt.core.manipulation.ChangeCorrectionProposalCore.getChange(ChangeCorrectionProposalCore.java:153)
	at org.eclipse.jdt.ls.core.internal.handlers.CodeActionResolveHandler.resolve(CodeActionResolveHandler.java:49)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$15(JDTLanguageServer.java:793)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
```

--- a/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java
+++ b/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java
@@ -302,7 +302,6 @@ class JavacCompilationUnitResolver implements ICompilationUnitResolver {
 			var fileObject = fileManager.getJavaFileObject(sourceUnitPath);
 			fileManager.cache(fileObject, CharBuffer.wrap(sourceUnit.getContents()));
 			AST ast = createAST(compilerOptions, apiLevel, context);
-			ast.setDefaultNodeFlag(ASTNode.ORIGINAL);
 			CompilationUnit res = ast.newCompilationUnit();
 			result.put(sourceUnit, res);
 			filesToUnits.put(fileObject, res);
@@ -337,6 +336,8 @@ class JavacCompilationUnitResolver implements ICompilationUnitResolver {
 				}
 				CompilationUnit res = result.get(sourceUnits[i]);
 				AST ast = res.ast;
+				int savedDefaultNodeFlag = ast.getDefaultNodeFlag();
+				ast.setDefaultNodeFlag(ASTNode.ORIGINAL);
 				JavacConverter converter = new JavacConverter(ast, javacCompilationUnit, context, rawText);
 				converter.populateCompilationUnit(res, javacCompilationUnit);
 				attachComments(res, context, fileObjects.get(i), converter, compilerOptions);
@@ -356,6 +357,7 @@ class JavacCompilationUnitResolver implements ICompilationUnitResolver {
 				ast.setBindingResolver(new JavacBindingResolver(javaProject, task, context, converter));
 				//
 				ast.setOriginalModificationCount(ast.modificationCount()); // "un-dirty" AST so Rewrite can process it
+				ast.setDefaultNodeFlag(savedDefaultNodeFlag);
 			}
 		} catch (IOException ex) {
 			ILog.get().error(ex.getMessage(), ex);

